woocommerce/woocommerce-ios
# [HACK week] Product list - Add Favorites section

Part of: #12274


## Description
List favorite products under a new section in product list table view. 

Changes
- Limit the max number of products that can be marked as favorite to avoid taking too much space in product list screen. In future, we can consider making "Favorites" section collapsible to make space for the "All" section.
- Load the favorites product IDs from remote and show the products under "Favorites" section.
- Observe to user defaults and reload when a product is marked/removed as favorite. 

## Testing instructions
- Login into the app
- Navigate to Products tab
- Open any product
- Tap "…" from the action sheet tap "Mark as favorite"
- Navigate back to the product list screen.
- You should see the product under "Favorites" section
- Open the same product again
- Tap "…" -> "Remove from a favourite"  on it to remove the product from the favourite list.
- Navigate back to the product list screen and validate that the "Favorites" section is updated.
- Mark a few products to the favorites section.
- Switch stores and ensure that the products marked as favorite are still marked as favorite. 
- Log out and log into the same store again. Validate that the favorite products list is cleared after logout.

## Screenshots

![Simulator Screen Recording - iPhone 15 Pro - 2024-03-15 at 20 16 26](https://github.com/woocommerce/woocommerce-ios/assets/524475/e013286c-f0e8-4aa2-bb79-dce8f7a48256)


---
